### PR TITLE
Make SummarySection more declarative

### PR DESCRIPTION
**Pull request by mechaffin**
# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [x] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

Make SummarySection more declarative

### CODE REVIEW

File: src/components/pages/devices/flyouts/deviceDelete/deviceDelete.js [hunk 1]
```diff
@@ -129,11 +131,14 @@ export class DeviceDelete extends Component {
...
-              <SummaryCount>{summaryCount}</SummaryCount>
-              <SectionDesc>{summaryMessage}</SectionDesc>
-              {this.state.isPending && <Indicator />}
-              {completedSuccessfully && <Svg className="summary-icon" path={svgs.apply} />}
+            <SummarySection>
+              <SectionHeader>{t('devices.flyouts.delete.summaryHeader')}</SectionHeader>
+              <SummaryBody>
+                <SummaryCount>{summaryCount}</SummaryCount>
+                <SectionDesc>{summaryMessage}</SectionDesc>
+                {this.state.isPending && <Indicator />}
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDelete/deviceDelete.js]: You should confirm with design but I believe the goal is the summary section indicators should be bar patterned.

File: src/components/pages/rules/flyouts/ruleStatus/ruleStatus.js [hunk 2]
```diff
@@ -87,25 +89,30 @@ export class RuleStatus extends Component {
             {
               rules.map((rule) => (
                 <SummarySection key={rule.id} title={rule.name} className="padded-bottom">
-                  <FormLabel className="rule-description">{rule.description}</FormLabel>
-                  <SummaryCount>{rule.count && rule.count.response ? rule.count.response : '---'}</SummaryCount>
-                  <SectionDesc>{t('rules.flyouts.ruleEditor.devicesAffected')}</SectionDesc>
+                  <SectionHeader>{rule.description}</SectionHeader>
+                  <SummaryBody>
+
```
Comment (spryor) [inline src/components/pages/rules/flyouts/ruleStatus/ruleStatus.js]: newline

**Comment by spryor** [comment]
Also build failure

File: src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js [hunk 3]
```diff
@@ -394,17 +396,30 @@ export class RuleEditor extends LinkedComponent {
...
+              - Also, allow for additional changes to be made while the flyout is open.
+              */
+            }
+          </SummaryBody>
         </SummarySection>
+
         {
-          error && <AjaxError t={t} error={error} />
+          error &&
+          <AjaxError t={t} error={error} />
```
Comment (spryor) [inline src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js]: leave this as oneline

File: src/components/pages/devices/flyouts/deviceNew/deviceNew.js [hunk 4]
```diff
@@ -391,11 +392,22 @@ export class DeviceNew extends LinkedComponent {
...
-              {completedSuccessfully && <Svg className="summary-icon" path={svgs.apply} />}
+            <SummarySection>
+              <SectionHeader>{t('devices.flyouts.new.summaryHeader')}</SectionHeader>
+              <SummaryBody>
+                <SummaryCount>{summaryCount || 0}</SummaryCount>
+                <SectionDesc>{summaryMessage}</SectionDesc>
+                {this.state.isPending && <Indicator />}
+                {completedSuccessfully && <Svg className="summary-icon" path={svgs.apply} />}
+                {
+                      /*
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceNew/deviceNew.js]: The indentation doesn't match the one below

File: src/components/pages/rules/flyouts/ruleStatus/ruleStatus.js [hunk 5]
```diff
@@ -86,27 +88,39 @@ export class RuleStatus extends Component {
...
-                !isPending && <BtnToolbar>
-                  <Btn svg={svgs.apply} primary={true} type="submit">{t('rules.flyouts.ruleEditor.apply')}</Btn>
-                  <Btn svg={svgs.cancelX} onClick={onClose}>{t('rules.flyouts.ruleEditor.cancel')}</Btn>
-                </BtnToolbar>
-              }
-            </SummarySection>
+
+            {
+              error &&
+              <AjaxError t={t} error={error} />
```
Comment (spryor) [inline src/components/pages/rules/flyouts/ruleStatus/ruleStatus.js]: oneline

File: src/components/shared/forms/styles/select.scss [hunk 6]
```diff
@@ -442,30 +351,118 @@ $select-padding-horizontal: 8px;
       // move the baseline up by 1px
       padding: ($select-item-padding-vertical - 1) $select-item-padding-horizontal ($select-item-padding-vertical + 1);
       @include border-left-radius( $select-item-border-radius );
-      @include themify($themes) {
+    }
+  }
+
+  .Select--multi.is-disabled .Select-value-icon {
```
Comment (spryor) [inline src/components/shared/forms/styles/select.scss]: oneline

File: src/components/shared/forms/styles/select.scss [hunk 7]
```diff
@@ -442,30 +351,118 @@ $select-padding-horizontal: 8px;
...
+      .Select-value-label { color: themed('colorSelectText'); }
+
+      a.Select-value-label {
+        &:hover,
+        &:focus { color: themed('colorSelectBorderColor');
+        }
+      }
+    }
+
+    .Select-input {
```
Comment (spryor) [inline src/components/shared/forms/styles/select.scss]: `.Select-input > input { color: themed('colorSelectText'); }`

File: src/components/shared/forms/styles/select.scss [hunk 8]
```diff
@@ -442,30 +351,118 @@ $select-padding-horizontal: 8px;
...
+    .is-focused:not(.is-open) > .Select-control { border-color: themed('colorSelectBorderColor'); }
+
+    .Select-placeholder,
+    .Select--single > .Select-control .Select-value {
+        color: themed('colorSelectPlaceholderText');
+    }
+
+    .has-value.Select--single > .Select-control .Select-value,
+    .has-value.is-pseudo-focused.Select--single > .Select-control .Select-value {
+      .Select-value-label { color: themed('colorSelectText'); }
```
Comment (spryor) [inline src/components/shared/forms/styles/select.scss]: Is `.Select-value-label` ever not an `a` tag? If not these can be merged.

File: src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js (hunk 3 above)
Comment (mechaffin) [reply src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js]: I was making these match the error display in devices... so now, I'll go change the devices ones too

**Pull request merged**